features/support/helpers.rb

```diff
@@ -44,6 +44,36 @@ def stub_request_with_address(address, body = nil)
   end
 end
 
+
+PoltergeistProc = -> (page, key, value) do
+  page.driver.set_cookie(key, value)
+end
+
+SeleniumProc = -> (page, key, value) do
+  page.driver.browser.manage.add_cookie(name: key, value: value)
+end
+
+RackProc = -> (_page, key, value) do
+  headers = {}
+  Rack::Utils.set_cookie_header!(headers, key, value)
+  cookie_string = headers['Set-Cookie']
+  Capybara.current_session.driver.browser.set_cookie(cookie_string)
+end
+
+WebkitProc = -> (_page, key, value) do
+  headers = {}
+  Rack::Utils.set_cookie_header!(headers, key, value)
+  cookie_string = headers['Set-Cookie']
+  Capybara.current_session.driver.browser.set_cookie(cookie_string)
+end
```

Comment:
Nice idea @johnnymo87, but I saw that we are using a Hash to get the correct proc, why not set the value of RackTest driver and WebKit driver to the same proc for example RackProc ?